lib/vsc/filesystem/quota/ktools.py

```diff
@@ -0,0 +1,349 @@
+            self.payload[storage_name] = []
+
+    def push_quota(self, owner, fileset, quota, shared=False):
+        """
+        Push quota to accountpage: it belongs to owner (can either be user_id or vo_id),
+        in the given fileset and quota.
+        :param owner: the name of the user or VO to which the quota belongs
+        :param fileset: fileset name
+        :param quota: actual quota data
+        :param shared: is this a shared user/VO quota or not?
+        """
+        params = {
+            "fileset": fileset,
+            "used": quota.used,
+            "soft": quota.soft,
+            "hard": quota.hard,
+            "doubt": quota.doubt,
+            "expired": quota.expired[0],
+            "remaining": quota.expired[1] or 0,  # seconds
+            "files_used": quota.files_used,
+            "files_soft": quota.files_soft,
+            "files_hard": quota.files_hard,
+            "files_doubt": quota.files_doubt,
+            "files_expired": quota.files_expired[0],
+            "files_remaining": quota.files_expired[1] or 0,  # seconds
+        }
+
+        if self.kind == UsageType.UserUsage:
+            params['user'] = owner
+        elif self.kind == UsageType.FilesetUsage:
+            params['vo'] = owner
+
+        if shared:
+            self.push(self.storage_name_shared, params)
+        else:
+            self.push(self.storage_name, params)
+
+    def _push(self, storage_name, payload):
+        """Does the actual pushing to the REST API"""
+
+        if self.dry_run:
+            logging.info("Would push payload to account web app: %s", payload)
+        else:
+            try:
+                cl = self.client.usage.storage[storage_name]
+                if self.kind == UsageType.UserUsage:
+                    logging.debug("Pushing user payload to account web app: %s", payload)
+                    cl = cl.user
+                elif self.kind == UsageType.FilesetUsage:
+                    logging.debug("Pushing vo payload to account web app: %s", payload)
+                    cl = cl.vo
+                else:
+                    logging.error("Unknown quota kind, not pushing any quota to the account page")
+                    return
+                cl.size.put(body=payload)  # if all is well, there's nothing returned except (200, empty string)
+            except Exception:
+                logging.error("Could not store quota info in account web app")
+                raise
+
+
+def map_uids_to_names():
+    """Determine the mapping between user ids and user names."""
+    ul = pwd.getpwall()
+    d = {}
+    for u in ul:
+        d[u[2]] = u[0]
+    return d
+
+
+def determine_grace_period(grace_string):
+    grace = GPFS_GRACE_REGEX.search(grace_string)
+    nograce = GPFS_NOGRACE_REGEX.search(grace_string)
+
+    if nograce:
+        expired = (False, None)
+    elif grace:
+        grace = grace.groupdict()
+        grace_time = 0
+        if grace['days']:
```

Comment:
isn't this logic something that gpfsbeat should do?

Reply:
No, I don't think so.

Reply:
but now you have to redo this logic on every spot that wants to know the grace period?

Why not compute it in it seconds at the source? It will also makes things easier in kibana?

Reply:
I don't really care about this in kibana, tbh. But PRs welcome :p